pkg/resolution/resolver/git/config.go

```diff
@@ -43,3 +51,54 @@ const (
 	// APISecretNamespaceKey is the config map key for the token secret's namespace
 	APISecretNamespaceKey = "api-token-secret-namespace"
 )
+
+type GitResolverConfig struct {
+	ScmTokens map[string]ScmInfo
+}
+
+type ScmInfo struct {
+	Timeout            string `json:"fetch-timeout"`
+	URL                string `json:"default-url"`
+	Revision           string `json:"default-revision"`
+	Org                string `json:"default-org"`
+	ServerURL          string `json:"server-url"`
+	SCMType            string `json:"scm-type"`
+	APISecretName      string `json:"api-token-secret-name"`
+	APISecretKey       string `json:"api-token-secret-key"`
+	APISecretNamespace string `json:"api-token-secret-namespace"`
+}
+
+func GetGitConfig(ctx context.Context) GitResolverConfig {
+	var scmInfo interface{} = &ScmInfo{}
+	structType := reflect.TypeOf(scmInfo).Elem()
+	gitResolverConfig := GitResolverConfig{ScmTokens: map[string]ScmInfo{}}
+	conf := framework.GetResolverConfigFromContext(ctx)
+	for k, v := range conf {
+		var tokenIdentifier, tokenKey string
+		key := strings.Split(k, ".")
+		if len(key) >= 3 && key[0] == "identifier" {
```

Comment:
We don't support having a "." in the name of a configuration item, we should look for `len(key)== 3`.

As mentioned before, I think the `identifier` here is redundant. This could be something like:

```golang
var configName
key := strings.Split(k, ".")
switch len(key) {
case 2:
  configName = key[0]
case 1:
  configName = "default"
default:
  // return an error
}
```

Reply:
ack length should be 3, but i just added if someone is adding by mistake, so it is processed but it will not match any jsonTag

Reply:
I used `2` thinking ahead about my previous comment that the `identifier` part is redundant.
The non-match will be silently ignored, which is not ok I think.

Reply:
handled this and returning error for other cases